codec-http2/src/test/java/io/netty/handler/codec/http2/Http2CodecTest.java

```diff
@@ -0,0 +1,159 @@
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertTrue;
+
+/**
+ * Unit tests for {@link Http2Codec}.
+ */
+public class Http2CodecTest {
+
+    private static final EventLoopGroup group = new DefaultEventLoop();
+    private Channel serverChannel;
+    private Channel clientChannel;
+    private LastInboundHandler serverLastInboundHandler;
+
+    @Before
+    public void setUp() throws InterruptedException {
+        LocalAddress serverAddress = new LocalAddress(getClass().getName());
+        serverLastInboundHandler = new SharableLastInboundHandler();
+        ServerBootstrap sb = new ServerBootstrap()
+                .channel(LocalServerChannel.class)
+                .group(group)
+                .childHandler(new Http2Codec(true, serverLastInboundHandler));
+        serverChannel = sb.bind(serverAddress).sync().channel();
+
+        Bootstrap cb = new Bootstrap()
+                .channel(LocalChannel.class)
+                .group(group)
+                .handler(new Http2Codec(false, new TestChannelInitializer()));
+        clientChannel = cb.connect(serverAddress).sync().channel();
+    }
+
+    @AfterClass
+    public static void shutdown() {
+        group.shutdownGracefully();
+    }
+
+    @After
+    public void tearDown() throws Exception {
+        clientChannel.close().sync();
+        serverChannel.close().sync();
+    }
+
+    @Test
+    public void multipleOutboundStreams() throws InterruptedException{
+        Http2StreamChannelBootstrap b = new Http2StreamChannelBootstrap();
+        b.parentChannel(clientChannel).handler(new TestChannelInitializer());
+
+        Channel childChannel1 = b.connect().sync().channel();
+        assertTrue(childChannel1.isActive());
+        assertFalse(((AbstractHttp2StreamChannel) childChannel1).hasStreamId());
+        Channel childChannel2 = b.connect().channel();
+        assertTrue(childChannel2.isActive());
+        assertFalse(((AbstractHttp2StreamChannel) childChannel2).hasStreamId());
+
+        Http2Headers headers1 = new DefaultHttp2Headers();
+        Http2Headers headers2 = new DefaultHttp2Headers();
+        // Test that streams can be made active (headers sent) in different order than the corresponding channels
+        // have been created.
+        childChannel2.writeAndFlush(new DefaultHttp2HeadersFrame(headers2));
+        childChannel1.writeAndFlush(new DefaultHttp2HeadersFrame(headers1));
+
+        Http2HeadersFrame headersFrame2 = serverLastInboundHandler.blockingReadInbound();
+        assertNotNull(headersFrame2);
+        assertEquals(3, headersFrame2.streamId());
+
+        Http2HeadersFrame headersFrame1 = serverLastInboundHandler.blockingReadInbound();
+        assertNotNull(headersFrame1);
+        assertEquals(5, headersFrame1.streamId());
+
+        assertTrue(((AbstractHttp2StreamChannel) childChannel2).hasStreamId());
+        assertTrue(((AbstractHttp2StreamChannel) childChannel2).hasStreamId());
```

Comment:
A few requests :)

1. can we get rid of the `hasStreamId` method on this class and instead just use `isStreamIdValid`?

2. lets add the following:

```java
assertEquals(3, ((AbstractHttp2StreamChannel) childChannel2).getStreamId());
assertEquals(5, ((AbstractHttp2StreamChannel) childChannel1).getStreamId());
```

`3`. change the `getStreamId()` to `streamId()` for consistency.